mozilla/fx-private-relay
# MPP-3687 [Android] Delete mask modal not appearing always

This PR fixes MPP-3687  

# New feature description

[Android] Delete mask modal not appearing always and deleting the mask immediately. This adds a check so that the `onConfirm` function checks if the modal is open. 

```Typescript
const onConfirm: FormEventHandler = (event) => {
    event.preventDefault();

    if (modalState.isOpen) {
      props.onDelete();
      modalState.close();
    }
  };
```

# Screenshot (if applicable)

Not applicable.

# How to test
 
1. Navigate to the Relay email mask dashboard;
2. Generate a new email mask;
3. Delete the email mask;
4. Repeat steps 2-3 a few times and observe behaviour;

**Expected result:**

The delete mask modal appears whenever a user tries to delete a mask;

**Incorrect result:**

Intermittently, the modal does not appear and the mask is deleted instantly;

# Checklist (Definition of Done)
